src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs

```diff
@@ -1,12 +1,99 @@
+            if (options == null) throw new ArgumentNullException(nameof(options));
+
+            services.AddSingleton<IScheduledTaskOptions<TScheduledTask>>(options);
+            return options;
+        }
+
+        public static IScheduledTaskOptions<TScheduledTask> AddScheduledTaskOptions<TScheduledTask>(
+            this IServiceCollection services,
+            Action<IScheduledTaskOptions<TScheduledTask>> configureOptions
+        )
+            where TScheduledTask : class, IScheduledTask
+        {
+            if (services == null) throw new ArgumentNullException(nameof(services));
+            if (configureOptions == null) throw new ArgumentNullException(nameof(configureOptions));
+
+            var options = new ScheduledTaskOptions<TScheduledTask>();
+            configureOptions.Invoke(options);
+            services.AddSingleton<IScheduledTaskOptions<TScheduledTask>>(options);
+            return options;
+        }
+
+        public static IServiceCollection AddScheduledTask<TScheduledTask>(
+            this IServiceCollection services
+        )
+            where TScheduledTask : class, IScheduledTask
+        {
+            if (services == null) throw new ArgumentNullException(nameof(services));
+            return AddService<TScheduledTask>(services);
+        }
+
+        public static IServiceCollection AddScheduledTask<TScheduledTask>(
+            this IServiceCollection services,
+            IScheduledTaskOptions<TScheduledTask> options
+        )
+            where TScheduledTask : class, IScheduledTask
+        {
+            if (services == null) throw new ArgumentNullException(nameof(services));
+            if (options == null) throw new ArgumentNullException(nameof(options));
+
+            services.AddSingleton<IScheduledTaskOptions<TScheduledTask>>(options);
+            return AddService<TScheduledTask>(services);
+        }
+
+        public static IServiceCollection AddScheduledTask<TScheduledTask>(
+            this IServiceCollection services,
+            System.Action<IScheduledTaskOptions<TScheduledTask>> configureOptions
+        )
+            where TScheduledTask : class, IScheduledTask
+        {
+            if (services == null) throw new ArgumentNullException(nameof(services));
+            if (configureOptions == null) throw new ArgumentNullException(nameof(configureOptions));
+
+            var options = new ScheduledTaskOptions<TScheduledTask>();
+            configureOptions.Invoke(options);
+            services.AddSingleton<IScheduledTaskOptions<TScheduledTask>>(options);
+            return AddService<TScheduledTask>(services);
+        }
+
+        private static IServiceCollection AddService<TScheduledTask>(
+            IServiceCollection services
+        )
+            where TScheduledTask : class, IScheduledTask
+        {
+            services.TryAddSingleton<IHostedService, SchedulerHostedService>();
```

Comment:
That's wrong. You need your own Interface

Reply:
implemented in fc33a7bb9de95152947d72badacc9cc9a33ace6e